tests/test_omegaconf.py

```diff
@@ -518,3 +518,93 @@ def test_resolve(cfg: Any, expected: Any) -> None:
 def test_resolve_invalid_input() -> None:
     with raises(ValueError):
         OmegaConf.resolve("aaa")  # type: ignore
+
+
+@mark.parametrize(
+    ("cfg_params,clear_resolver_params,expected"),
+    [
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR1",
+                resolver=lambda x: str(x).lower(),
+                use_cache=False,
+                replace=True,
+            ),
+            # clear_resolver_params
+            dict(name="TR1"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR2",
+                resolver=lambda x: str(x).upper(),
+                use_cache=False,
+                replace=True,
+            ),
+            # clear_resolver_params
+            dict(name="TR2"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params
+            dict(
+                name="TR3",
+                resolver=lambda x: str(x).upper(),
+                use_cache=False,
+                replace=False,
+            ),
+            # clear_resolver_params
+            dict(name="TR3"),
+            # expected
+            dict(pre_addition=False, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params: Check with a default resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="oc.env"),
+            # expected
+            dict(pre_addition=True, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params: Check with a default resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="oc.create"),
+            # expected
+            dict(pre_addition=True, post_addition=True, post_removal=False),
+        ),
+        # tuple
+        (
+            # cfg_params: Check with a nonexisting resolver
+            dict(),
+            # clear_resolver_params
+            dict(name="abcdef"),
+            # expected
+            dict(pre_addition=False, post_addition=False, post_removal=False),
+        ),
+    ],
+)
+def test_clear_resolver(
+    cfg_params: Any, clear_resolver_params: Any, expected: Any
+) -> None:
+
+    OmegaConf.clear_resolvers()
+    name = clear_resolver_params.get("name")
+    assert expected["pre_addition"] == OmegaConf.has_resolver(name)
+    if cfg_params:
+        OmegaConf.register_new_resolver(**cfg_params)
+        assert expected["post_addition"] == OmegaConf.has_resolver(name)
+
+    assert OmegaConf.clear_resolver(**clear_resolver_params)
+
+    assert expected["post_removal"] == OmegaConf.has_resolver(name)
```

Comment:
In all of the test cases, `expected["post_removal"]` is `False`. I think we should just use `assert not OmegaConf.has_resolver(name)` here.

Reply:
I had a sense perhaps some of the test cases will need change, as first I wanted to make sure the implementation of clear_resolver seems okay. Let me slim the tests down and then I think it will end up something like you suggested.

Reply:
Done!